odoo/orm/fields.py

```diff
@@ -1699,9 +1715,10 @@ def __get__(self, record: BaseModel, owner=None) -> T:
         if self.compute and self.store:
             # process pending computations
             self.recompute(record)
+            # cache may be changed by recompute
+            field_cache = self._get_cache(env)
```

Comment:
Why can the cache be changed by recompute?  Is it because x2many computed fields may indirectly invoke `unlink()`, which invalidates `transaction.field_cache` from the root?